paper.md

```diff
@@ -0,0 +1,257 @@
+# there are also caching options for the s3 middleware, not shown here
+storage_options = {
+    'anon': True,
+    'client_kwargs': {'endpoint_url': "https://s3server.ac.uk"}
+}
+fs = s3fs.S3FileSystem(**storage_options)
+file_uri = "s3-bucket/myfile.nc"
+with fs.open(file_uri, 'rb') as s3_file:
+    nc = pyfive.File(s3_file)
+    dataset = nc[var]
+```
+
+for an HTTPS data server, the usage is similar:
+
+```python
+import fsspec
+import pyfive
+# there are also caching options for the fsspec middleware, not shown here
+fs = fsspec.filesystem('http')
+with fs.open("https://site.com/myfile.nc", 'rb') as http_file:
+    nc = pyfive.File(http_file)
+    dataset = nc[var]
+```
+
+This is of course exactly the same pattern as remote access using `h5py`, and that is by design - to make
+moving to `pyfive` easy for users!
+
+## Lazy Loading
+
+A key tenet of efficient remote access is that variable inspection is quick and involves the minimimum of network
+traffic between storage and the client application.  
+However when this is coupled with the common pattern of 
+using Dask, some flexibility in what is loaded when is beneficial.
+
+By default when one inspects the contents of a file using `pyfive v1.0` nothing more is read from the file than
+the names of the variables ("datasets" in the language of HDF5): for example:
+```python
+with pyfive.File('myfile.h5','r') as f:
+  variables_in_file = [v for v in f]
+```
+involves nothing more than getting a set of variable names. When one wishes to inspect these variables:
+``` python
+with pyfive.File('myfile.h5','r') as f:
+  temp = f['temp']
+print(temp[1:10])
+```
+the default in `pyfive v1.0` is to get only the metadata associated with each variable - but crucially at 
+this point the b-tree index is also loaded and the variable can now be accessed outside the context
+manager. 
+Data loading is now completely lazy and the variable instance (`temp`) has all the information
+needed to extract data as needed. 
+This is done so that in Dask applications, when one passes
+each Dask computational chunk a portion of the variable, each such Dask chunk has already got the 
+index, and when it does want to load data, it can be as efficient as possible.
+
+However, there are situations where loading the b-tree at variable instantation is not wanted, and all
+is wanted is to be able to view all the variable attributes. 
+To support this option `pyfive` also offers the `get_lazy_view` file method, so one can do:
+``` python
+with pyfive.File('myfile.h5','r') as f:
+  temp = f.get_lazy_view('temp')
+print(temp.attrs())
+```
+
+It is still possible to access the data and the b-tree index is loaded when data access is first attempted.
+
+This extra lazy view is new functionality. 
+It would obviously have been possible to make the default not load the b-tree, but in the opinion of the
+current `pyfive` maintainers loading the b-tree at variable instantiation is likely more consistent
```

Comment:
```suggestion
It would obviously have been possible to make the default not load the internal file metadata, but in the opinion of the
current `pyfive` maintainers, loading the internal file metadata during variable instantiation is likely more consistent
```